sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -299,6 +299,29 @@ public Instant getCurrentTimestamp() throws NoSuchElementException {
     return curTimestamp;
   }
 
+  @Override
+  public byte[] getCurrentRecordId() throws NoSuchElementException {
+    if (!this.offsetBasedDeduplicationSupported) {
+      throw new RuntimeException("UnboundedSource must enable offset-based deduplication.");
+    }
+    if (curRecord != null) {
+      return KafkaIOUtils.OffsetBasedDeduplication.getUniqueId(
+          curRecord.getTopic(), curRecord.getPartition(), curRecord.getOffset());
```

Comment:
Hm, I'm confused by the error (despite null check beforehand) and proper way to handle it. Is it due to accessing curRecord multiple times with only one null check? The error occurs on 2nd and 3rd ref, but not the 1st.

/Users/runner/work/beam/beam/sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java:309: error: [dereference.of.nullable] dereference of possibly-null reference curRecord
> Task :sdks:java:io:kafka:compileJava
          curRecord.getTopic(), curRecord.getPartition(), curRecord.getOffset());

Reply:
From an internal static analysis FAQ, it suggested:
1. Switch to the MonotonicNonNull annotation (if it applies).
2. Use a local reference.

I think option 1 applies and is better than local ref. So trying that out.